### ISSUE TITLE
[Feature request] Auto-activate "In selection" in replace dialog when text is selected

### ISSUE THREAD

**Issue by SpeechScientist** [notepad-plus-plus/notepad-plus-plus#4994]
### Description of the Issue
Proposal: if the Replace dialog is called while a text selection is active, "In Selection" should be activated by default

### Steps to reproduce
1. select some text
2. hit ctrl+h, or select Replace in the Search menu
3. if you want to replace something only in the selected part, you have to tick the "In selection" check box. 

Since you did select the text in the first place, I think it is reasonable to assume that doing the replace in the selected part only is precisely what you want to do, so I'm proposing that "In selection" in the Replace dialog should be active by default in this case.

**Comment by sasumner** [comment]
A much more frequent use-case is that you want the text that you have selected to appear in the Find-what zone when you hit ctrl+h.  Thus I down-vote this feature request...

BTW, it already works like your request when the amount of selected text is above a certain amount (but I forget exactly what the threshold amount is!).

**Comment by mcsilver** [comment]
Perhaps a compromise would allow both sets of functionality: add a setting/preference to allow the user to choose between the two. 

Also, or alternatively, perhaps 95% of the users who want the selected text to become the _Find what_ text only want it when the selected text contains no end-of-line characters (the selected text is not multi-line) and therefore used to determine whether to observe behaviour A or B. 

In other words:

1. User selects some-text
2. User presses Ctrl+H (Replace)
3. If some-text is multi-line, then
         automatically check _In selection_
         do _not_ add some-text to _Find what_
    else
         automatically uncheck _In selection_
         add some-text to _Find what_

I actually would like the same functionality for _Find_, not just for _Replace_.

**Comment by sasumner** [comment]
Duplicate of #3985 ; this one has a better issue title.

**Comment by mcsilver** [comment]
Agreed, this ticket has a better title. I wouldn't want to lose the good, thoughtful considerations already mentioned in #3985 though.

**Comment by mcsilver** [comment]
After having read what is in #3985, I would also like to add...

Another reason that I lean toward having **In Selection** automatically checked when at least one character is selected, or at least when the selection contains an EOL (a multiline selection), is that it would then work more like Visual Studio and Microsoft Word, which I am very used to. Is there an argument to be made for NPP working like other IDE's and editors, or am I unaware that there is an equal or greater number of them out there that behave in the opposite way?

**Comment by mcsilver** [comment]
Another solution is to turn these choices into Settings > Preferences. Not sure which section. Currently, only Miscellaneous or a new Find and Replace section make sense to me.

**Combo box**
label: "Automatically check In Selection for:"
options: "Any selected text", "Multi-line selected text", "Never"

**Combo box**
label: "Automatically populate Find with selected text for:"
options: "Any selected text", "Multi-line selected text", "Never"

Or you could just save yourself the hassle and do it the way _I_ want ;).

**Comment by sasumner** [comment]
>Another solution is to turn these choices into Settings > Preferences. Not sure which section. Currently, only Miscellaneous or a new Find and Replace section make sense to me.

See #8131

**Comment by sasumner** [comment]
>I actually would like the same functionality for Find

Not quite sure what you mean with this, but perhaps #8133 is of interest?

**Comment by mcsilver** [comment]
> > I actually would like the same functionality for Find
> 
> Not quite sure what you mean with this, but perhaps #8133 is of interest?

Yes, thanks for point this one out. The functionality described in #8133 _is_ desirable to me--to find occurrences only within the selected text--but what I meant above was for the **In Selection** checkbox to be auto-checked when there is selected text (or perhaps when the selected text is multi-line--I am flexible that way).
